cameraServer.py

```diff
@@ -0,0 +1,215 @@
+        rects.sort(key=lambda x:x[2])
+        
+        pairs = []
+        leftRect = None
+        for i, rect in enumerate(sorted(rects, key=lambda x:x[0])): #Get rectangle pairs
+            if rect[2] < (leftAngleSize + angleOffset) and rect[2] > (leftAngleSize - angleOffset):
+                leftRect = rect
+            elif rect[2] < (rightAngleSize + angleOffset) and rect[2] > (rightAngleSize - angleOffset):
+                if leftRect:
+                    pairs.append((leftRect, rect))
+                    leftRect = None
+                    
+        if len(pairs) >= 1:
+            closestToMiddle = min(pairs, key = lambda x:abs(x[0][0][0] - screenSize[0]/2))
+        else:
+            return None, None, img
+
+        circleContours = list(np.int0(cv2.boxPoints(closestToMiddle[0])))
+        circleContours.extend(list(np.int0(cv2.boxPoints(closestToMiddle[1]))))
+        circleContours = np.array(circleContours)
+        (x,y),radius = cv2.minEnclosingCircle(circleContours)
+        radius = int(radius)
+        center = (int(x), int(y))
+
+        angle = (closestToMiddle[0][1][0] * closestToMiddle[0][1][1]) / (closestToMiddle[1][1][0] * closestToMiddle[1][1][1])
+        
+        if display == True: #Create the annotated display if display is True
+            for pair in pairs:
+                if pair == closestToMiddle:
+                    for tape in pair:
+                        img = cv2.drawContours(img, [np.int0(cv2.boxPoints(tape))], 0, (0, 255, 0), thickness=2)
+                    img = cv2.circle(img, center, radius, (0, 255, 0))
+                else:
+                    for tape in pair:
+                        img = cv2.drawContours(img, [np.int0(cv2.boxPoints(tape))], 0, (0, 0, 255))
+
+            return angle, -(((x/screenSize[0])*2)-1), img
+    return None, None, img
+
+def getGroundPos(img, sample=False): #TODO: Ground Tape Function
+    return None, None, None #Distance from x, distance from y, angle of line
+    
+#if False:
+if __name__ == "__main__":
+    from cscore import CameraServer, VideoSource
+    from networktables import NetworkTables
+    
+    if len(sys.argv) >= 2:
+        configFile = sys.argv[1]
+
+    # read configuration
+    if not readConfig():
+        sys.exit(1)
+
+    # start NetworkTables
+    NetworkTables.initialize(server='10.47.74.2')
+
+    
+    nt = NetworkTables.getTable('/vision')
+    entry = nt.getEntry('target_tape_error')
+    entry_angle = nt.getEntry('angle')
+
+    # start cameras
+    cameras = []
+    for cameraConfig in cameraConfigs:
+        cameras.append(startCamera(cameraConfig))
+
+    sink = cameras[0][0].getVideo(camera=cameras[0][1])
+    source = cameras[0][0].putVideo('cv', 320, 240)
+
+    frame = np.zeros(shape=(screenSize[1], screenSize[0], 3))
+    while True:
+        time1, frame = sink.grabFrameNoTimeout(image=frame)
+        angle, percent, image = getRetroPos(frame, True)
+
+        if not percent:
+            percent = 999
```

Comment:
Sending a real number when no angle exists is not ideal behaviour try `float("NaN")` instead